types/index.d.ts

```diff
@@ -322,6 +322,15 @@ declare module "miragejs/server" {
     /** A number of ms to artificially delay responses to this route. */
     timing?: number;
   }
+  
+  export interface ResourceOptions {
+    /** Whitelist of shorthand options */
+    only?: string[];
+    /** Exclude list of shorthand options */
+    except?: string[];
```

Comment:
We can be a bit more helpful here, and use a union of the allowed values.

The values allowed are `'index' | 'show' | 'create' | 'update' | 'delete'`